ggshield/sca/output/text_handler.py

```diff
@@ -0,0 +1,138 @@
+        :param prefix: (Optional, default = None) Adds a prefix at the
+            beginning of a line
+        :return: The formatted message to display
+        """
+        result_buf = StringIO()
+
+        result_buf.write(
+            file_info(
+                file_result.location,
+                # Compute number of incidents in the file
+                sum(
+                    len(pkg_version.vulns) for pkg_version in file_result.package_vulns
+                ),
+            )
+        )
+
+        incident_n = 1
+        for package_version in file_result.package_vulns:
+            for vulnerability in package_version.vulns:
+                result_buf.write(
+                    sca_incident_header(incident_n, package_version, prefix=prefix)
+                )
+
+                result_buf.write(sca_incident_severity_line(vulnerability))
+                result_buf.write(sca_incident_summary_line(vulnerability))
+                result_buf.write(sca_incident_fix_version_line(vulnerability))
+                result_buf.write(sca_incident_cve_ids(vulnerability))
+
+                incident_n += 1
+
+        return result_buf.getvalue()
+
+
+def no_sca_vulnerabilities() -> str:
+    """
+    Build a message if no IaC vulnerabilities were found.
+    """
+    return format_text("\nNo SCA vulnerability has been found\n", STYLE["no_secret"])
+
+
+def sca_incident_header(
+    issue_n: int,
+    package_version: SCAVulnerablePackageVersion,
+    prefix: Optional[str] = None,
+) -> str:
+    return "\n{}{} Incident {} ({}): {}@{}\n".format(
+        format_text(">>>", STYLE["detector_line_start"]),
+        "" if prefix is None else f" {prefix}:",
+        issue_n,
+        format_text("SCA", STYLE["detector"]),
+        format_text(package_version.package_full_name, STYLE["detector"]),
+        format_text(package_version.version, STYLE["detector"]),
+    )
+
+
+def sca_incident_severity_line(vulnerability: SCAVulnerability) -> str:
+    """
+    Returns the severity line, with associated style
+    """
+    if vulnerability.severity.lower() == "critical":
+        severity_string = "Critical"
+        style = STYLE["sca_vulnerability_critical"]
+    elif vulnerability.severity.lower() == "high":
+        severity_string = "High"
+        style = STYLE["sca_vulnerability_high"]
+    elif vulnerability.severity.lower() == "medium":
+        severity_string = "Medium"
+        style = STYLE["sca_vulnerability_medium"]
+    elif vulnerability.severity.lower() == "low":
+        severity_string = "Low"
+        style = STYLE["sca_vulnerability_low"]
+    else:  # In the other case, print `severity`
+        severity_string = vulnerability.severity
+        style = STYLE["sca_vulnerability_unknown"]
+
+    return "Severity: {}\n".format(
+        format_text(severity_string, style),
+    )
+
+
+def sca_incident_summary_line(vulnerability: SCAVulnerability) -> str:
+    return "Summary: {}\n".format(vulnerability.summary)
```

Comment:
I think it's safe to use f-strings (supported since Python 3.6 and used elsewhere in the codebase)